src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1240,33 +1259,71 @@ private List<EquipmentModificationInfos> execCreateLine(NetworkStoreListener lis
+                            .withFeederName1(lineCreationInfos.getConnectionName1() != null ? lineCreationInfos.getConnectionName1() : lineCreationInfos.getEquipmentId())
+                            .withFeederName2(lineCreationInfos.getConnectionName2() != null ? lineCreationInfos.getConnectionName2() : lineCreationInfos.getEquipmentId())
+                            .withDirection1(lineCreationInfos.getConnectionDirection1())
+                            .withDirection2(lineCreationInfos.getConnectionDirection2())
+                            .withPositionOrder1(position1)
+                            .withPositionOrder2(position2)
+                            .withBranchAdder(lineAdder).build();
+                    algo.apply(network, true, subReporter);
 
-                // Set Permanent Current Limits if exist
-                CurrentLimitsInfos currentLimitsInfos1 = lineCreationInfos.getCurrentLimits1();
-                CurrentLimitsInfos currentLimitsInfos2 = lineCreationInfos.getCurrentLimits2();
+                    // Set Permanent Current Limits if exist
+                    CurrentLimitsInfos currentLimitsInfos1 = lineCreationInfos.getCurrentLimits1();
+                    CurrentLimitsInfos currentLimitsInfos2 = lineCreationInfos.getCurrentLimits2();
+                    var line = getLine(network, lineCreationInfos.getEquipmentId());
 
-                if (currentLimitsInfos1 != null && currentLimitsInfos1.getPermanentLimit() != null) {
-                    myLine.newCurrentLimits1().setPermanentLimit(currentLimitsInfos1.getPermanentLimit()).add();
-                }
-                if (currentLimitsInfos2 != null && currentLimitsInfos2.getPermanentLimit() != null) {
-                    myLine.newCurrentLimits2().setPermanentLimit(currentLimitsInfos2.getPermanentLimit()).add();
+                    if (currentLimitsInfos1 != null && currentLimitsInfos1.getPermanentLimit() != null) {
+                        line.newCurrentLimits1().setPermanentLimit(currentLimitsInfos1.getPermanentLimit()).add();
+                    }
+                    if (currentLimitsInfos2 != null && currentLimitsInfos2.getPermanentLimit() != null) {
+                        line.newCurrentLimits2().setPermanentLimit(currentLimitsInfos2.getPermanentLimit()).add();
+                    }
+                } else if (voltageLevel1.getTopologyKind() == TopologyKind.NODE_BREAKER &&
+                        voltageLevel2.getTopologyKind() == TopologyKind.BUS_BREAKER) {
+                    addLineAndSubReporter(network, voltageLevel1, voltageLevel2, lineCreationInfos, true, false, subReporter);
+                } else {
+                    addLineAndSubReporter(network, voltageLevel1, voltageLevel2, lineCreationInfos, false, voltageLevel1.getTopologyKind() == TopologyKind.BUS_BREAKER &&
+                            voltageLevel2.getTopologyKind() == TopologyKind.NODE_BREAKER, subReporter);
                 }
```

Comment:
I think that just one 'else' is sufficient, with the two booleans 'withSwitch1' and 'withSwitch2' set to true ...
because withSwitch is only used when voltage level is in node breaker ...